dbt/include/firebolt/macros/adapters.sql

```diff
@@ -178,15 +178,23 @@
 
 
 {% macro firebolt__truncate_relation(relation) -%}
-    {% call statement('truncate_relation', fetch_result=True) -%}
-        SELECT COUNT(*) AS count
-          FROM information_schema.tables
-         WHERE "table_name" = '{{ relation }}';
-    {% endcall %}
-    {% set num_rows = load_result('truncate_relation').table %}
-    {% if num_rows == '1' %}
-        {% call statement('truncate_relation') %}
-            TRUNCATE {{ relation.identifier }}
-        {%- endcall %}
-    {% endif %}
+{# Firebolt doesn't currently support TRUNCATE, so DROP CASCADE.
+   Get table list from SHOW TABLES and save. After drop, recreate table. #}
+    {% call statement('list_tables', fetch_result=True) -%}
+        SHOW TABLES
+    {%- endcall %}
+    {% set tables = load_result('list_tables')['data'] %}
+    {% call statement('table_schema') -%}
+        DROP TABLE {{ relation.identifier }} CASCADE
+    {%- endcall %}
+    {# SHOW TABLES returns a dictionary. We want the list of tuples at key=='data'.
+       Unluckily, we have to step through each one to find the one that corresponds
```

Comment:
Can you use agate for this? I think I saw it used in other places

Reply:
You mean to find the right table, instead of using the loop?

Reply:
Yes -- I think there are examples in the core repo where they use agate to filter

Reply:
Took me a while, but the example was actually in our adapter code. Anyway, done now.